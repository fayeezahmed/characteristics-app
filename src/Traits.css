.traits {
    display: flex;
    align-items: center;
    height: 100%;
}

.traits li{
    flex: 1;
    background-color: lightblue;
    list-style: none;
    margin: .1em;
    padding: .3em;
    border-radius: 3px;
    margin: 1px;
    flex: column;
    height: 60px;
    min-width: 100px;
}
