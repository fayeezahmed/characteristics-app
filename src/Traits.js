import React, { Component, Fragment } from "react";
import './Traits.css';

var _ = require('lodash');
 
// https://www.quackit.com/html/html_editors/scratchpad/?example=/css/flexbox/examples/flexbox_12_column_grid_2
// https://www.quackit.com/css/flexbox/examples/
class Traits extends Component {
  constructor(props) {
    super(props)
    this.increaseCount = this.increaseCount.bind(this);
    this.createTraits = this.createTraits.bind(this);

    this.allTraits = this.props.allTraits;
    

    let combinedTraitsAndDays = _.zipObject(this.allTraits, _.map(this.allTraits, () => {
      return {
        Mon : 0, 
        Tues : 0,
        Wed : 0,
        Thurs : 0,
        Fri : 0,
        Sat : 0,
        Sun : 0}
    } ))
  
    this.state = combinedTraitsAndDays;
  }

  componentDidUpdate(){
    console.log(this.state);
  }

  increaseCount(trait, day){
    // console.log(trait, day)
    let traits = this.state;

   traits[trait][day] += 1
    this.setState = {traits};
    // console.log(this.state[trait]);
  }

  createTraits(trait) {
    // console.log(this)
    // TODO: Needs to be dynamically generated
    // TODO: Need to save state
    return <Fragment>
      <ul key= {trait.key + "_Group"}className="traits">
        <li key={trait.key} className="trait">
            {trait}
        </li>
        <li key={trait.key + "Mon"} onClick={() => this.increaseCount(trait, "Mon")}>
          {this.state[trait]["Mon"]}
        </li>
        <li key={trait.key + "Tues"} onClick={() => this.increaseCount(trait, "Tues")}>
        {this.state[trait]["Tues"]}
        </li>
        <li key={trait.key + "Wed"} onClick={() => this.increaseCount(trait, "Wed")}>
        {this.state[trait]["Wed"]}
        </li>
        <li key={trait.key + "Thurs"}onClick={() => this.increaseCount(trait, "Thurs")}>
        {this.state[trait]["Thurs"]}
        </li>
        <li key={trait.key + "Fri"} onClick={() => this.increaseCount(trait, "Fri")}>
        {this.state[trait]["Fri"]}
        </li>
        <li key={trait.key + "Sat"} onClick={() => this.increaseCount(trait, "Sat")}>
        {this.state[trait]["Sat"]}
        </li>
        <li key={trait.key + "Sun"} onClick={() => this.increaseCount(trait, "Sun")}>
        {this.state[trait]["Sun"]}
        </li>
      </ul> 
    </Fragment>       
  }
 
  render() {
    // const allTraits = this.props.allTraits;
    const listTraits = this.allTraits.map(this.createTraits);
    return (
      <div key="ALLTRAITS">
        {listTraits}
      </div>
    )
  }
};
 
 
export default Traits;