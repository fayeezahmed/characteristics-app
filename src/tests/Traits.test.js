import React from 'react';
import Traits from '../Traits';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const propTraits = ['Patience', 'Cheerfulness', 'Other'];
  const tree = renderer.create(<Traits allTraits={propTraits}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
