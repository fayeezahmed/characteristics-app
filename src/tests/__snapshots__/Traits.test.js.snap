// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ul
    className="traits"
  >
    <li
      className="trait"
    >
      Patience
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
  </ul>
  <ul
    className="traits"
  >
    <li
      className="trait"
    >
      Cheerfulness
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
  </ul>
  <ul
    className="traits"
  >
    <li
      className="trait"
    >
      Other
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
    <li
      onClick={[Function]}
    >
      0
    </li>
  </ul>
</div>
`;
